src/server/game/Entities/Creature/Creature.cpp

```diff
@@ -1483,9 +1525,12 @@ bool Creature::LoadCreatureFromDB(ObjectGuid::LowType spawnId, Map* map, bool ad
 
     m_respawnTime = GetMap()->GetCreatureRespawnTime(m_spawnId);
 
-    // Is the creature script objecting to us spawning? If yes, delay by one second (then re-check in ::Update)
-    if (!m_respawnTime && !sScriptMgr->CanSpawn(spawnId, GetEntry(), GetCreatureTemplate(), GetCreatureData(), map))
-        m_respawnTime = time(NULL)+1;
+    if (m_respawnCompatibilityMode)
+    {
+        // Is the creature script objecting to us spawning? If yes, delay by one second (then re-check in ::Update)
+        if (!m_respawnCompatibilityMode && !m_respawnTime && !sScriptMgr->CanSpawn(spawnId, GetEntry(), GetCreatureTemplate(), GetCreatureData(), map))
```

Comment:
possible I moved it to the wrong place on one of the many rebase operations. It should be outside that check.

I'll get to looking at this stuff maybe, this weekend.